app/lib/katello/resources/cdn.rb

```diff
@@ -1,7 +1,14 @@
 module Katello
   module Resources
     module CDN
-      SUPPORTED_SSL_VERSIONS = ['SSLv23', 'TLSv1'].freeze
+      SUPPORTED_TLS_VERSIONS_MAPPING = {
+        'TLSv1' => OpenSSL::SSL::TLS1_VERSION,
+        'TLSv1.1' => OpenSSL::SSL::TLS1_1_VERSION,
+        'TLSv1.2' => OpenSSL::SSL::TLS1_2_VERSION,
+        'TLSv1.3' => (OpenSSL::SSL::TLS1_3_VERSION if defined?(OpenSSL::SSL::TLS1_3_VERSION))
```

Comment:
Does this work? What value does it put into the hash?

Reply:
Yes, it works. If it's not defined, `defined?` returns `nil`, the "then" statement also evaluates to `nil`, and the corresponding KVP is removed by the `Hash#compact` method.

https://ruby-doc.org/2.7.8/Hash.html#method-i-compact
https://ruby-doc.org/2.7.8/syntax/miscellaneous_rdoc.html#label-defined-3F

The relevant behavior of modifier `if` wasn't documented at https://ruby-doc.org/2.7.8/syntax/control_expressions_rdoc.html , but `(expr if nil)` evaluates to `nil`:

```
irb(main):001:0> defined?(foo)
=> nil
irb(main):002:1* myhash = {
irb(main):003:1*   'one' => 'oneval',
irb(main):004:1*   'two' => ('twoval' if defined?(foo))
irb(main):005:0> }
=> {"one"=>"oneval", "two"=>nil}
irb(main):006:0> myhash.compact
=> {"one"=>"oneval"}
```